### ISSUE TITLE
Support configurable TLS Version for Custom Domains - AWS API Gateway

### ISSUE THREAD

**Issue by wandrew26** [terraform-providers/terraform-provider-aws#9100]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS announced configurable Transport Layer Security Version for Custom Domains in AWS API Gateway

### New or Affected Resource(s)

API Gateway

* aws_api_gateway_domain_name

### Potential Terraform Configuration

```
resource "aws_api_gateway_domain_name" "example" {
  domain_name              = "api.example.com"
  regional_certificate_arn = "${aws_acm_certificate_validation.example.certificate_arn}"
  security_policy          = "TLSv1.2"

  endpoint_configuration {
    types = ["REGIONAL"]
  }
}
```

### References

* https://aws.amazon.com/de/about-aws/whats-new/2019/06/amazon-api-gateway-adds-configurable-transport-layer-security-version-custom-domains/

**Comment by ewbankkit** [comment]
Requires [AWS SDK v1.20.7](https://github.com/aws/aws-sdk-go/releases/tag/v1.20.7) via https://github.com/terraform-providers/terraform-provider-aws/pull/9084 (once it bumps).

**Comment by ra-hardy** [comment]
Is #9084 bumped? (I'm not sure what that means!)

**Comment by ewbankkit** [comment]
@ra-hardy I think when I wrote that comment [Renovate Bot](https://github.com/apps/renovate) hadn't bumped the AWS SDK version in the then open PR. Anyway, the minimum required AWS SDK version is now merged; The functionality you required is implemented in https://github.com/terraform-providers/terraform-provider-aws/pull/9128 which has yet to be merged.

**Comment by bflad** [comment]
Support for the new `security_policy` argument for the `aws_api_gateway_domain_name` resource has been merged and will release with version 2.20.0 of the Terraform AWS Provider, later this week. 👍 Thanks to @stack72!
